vendor/src/github.com/docker/libnetwork/drivers/bridge/bridge.go

```diff
@@ -1269,7 +1270,7 @@ func (d *driver) ProgramExternalConnectivity(nid, eid string, options map[string
 	}
 
 	// Program any required port mapping and store them in the endpoint
-	endpoint.portMapping, err = network.allocatePorts(endpoint, network.config.DefaultBindingIP, d.config.EnableUserlandProxy)
+	endpoint.portMapping, err = network.allocatePorts(endpoint, network.config.DefaultBindingIP, d.config.EnableUserlandProxy, d.config.UserlandProxyPath)
```

Comment:
Please do not add this new parameter in all the function calls.
Just modify portmapper constructor to take theproxypath.